### ISSUE TITLE
Persist bounty filters in URL

### ISSUE THREAD

**Issue by john-light** [status-im/open-bounty#355]
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to copy a link so that visitors are taken directly to the search query I have filtered for on the open bounties page.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: I would like to be able to create a search filter e.g. owner: aragon on the open bounties page and then share this search filter with someone using a custom link. That way when someone clicks the link they see all of the bounties with that search filter without having to enter the search filter themselves.

#### Expected behavior

- Bounty filters should be represented in the URL as query params
- All current filtering combinations should be correctly encoded in the URI
- Loading a page with filters specified in the URL query params should cause the app to properly load those filters into the application state and update the UI accordingly.

(added by @martinklepsch)

#### Actual behavior

- Filters are not persisted in URL and reloading a page resets all filters.

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

- Enter a search filter e.g. owner: aragon
- Notice that the link remains static and does not update to reflect the search filter

### Solution

- Use [bide query params](https://github.com/funcool/bide#user-guide) to store filters in the application URL.
- Use anchor-style URLs as already done for other routes (i.e. `#/bounties?max-amount=400`).

**Comment by martinklepsch** [comment]
This should be easier once #323 is merged.

**Comment by status-open-bounty** [comment]
![Contract deploying](https://openbounty.status.im/img/deploying_contract.png)
Transaction: [0xa467d1fdf53c8ccaa5d2b512452797f0b51c553cc535915128c7a077e014b6fc](https://etherscan.io/tx/0xa467d1fdf53c8ccaa5d2b512452797f0b51c553cc535915128c7a077e014b6fc)

**Comment by msuess** [comment]
Taking a stab at this!

**Comment by status-open-bounty** [comment]
![Contract deploying](https://openbounty.status.im/img/deploying_contract.png)
Transaction: [0xc712cf31fff626b60185813c2d8cb49344c964bfe3bdb80e29d35d0d02c44fc2](https://etherscan.io/tx/0xc712cf31fff626b60185813c2d8cb49344c964bfe3bdb80e29d35d0d02c44fc2)
